src/main/java/org/opensearch/neuralsearch/search/util/HybridSearchSortUtil.java

```diff
@@ -0,0 +1,119 @@
+     * @param queryTopDocs list of top docs which have results with top scores.
+     * @return sort criteria
+     */
+    public static Sort evaluateSortCriteria(final List<QuerySearchResult> querySearchResults, final List<CompoundTopDocs> queryTopDocs) {
+        if (!checkIfSortEnabled(querySearchResults)) {
+            return null;
+        }
+        return createSort(getTopFieldDocs(queryTopDocs));
+    }
+
+    // Check if sort is enabled by checking docValueFormats Object
+    private static boolean checkIfSortEnabled(final List<QuerySearchResult> querySearchResults) {
+        if (querySearchResults == null || querySearchResults.isEmpty() || querySearchResults.get(0) == null) {
+            throw new IllegalArgumentException("shard result cannot be null in the normalization process.");
+        }
+        return querySearchResults.get(0).sortValueFormats() != null;
+    }
+
+    // Get the topFieldDocs array from the first shard result
+    private static TopFieldDocs[] getTopFieldDocs(final List<CompoundTopDocs> queryTopDocs) {
+        // loop over queryTopDocs and return the first set of topFieldDocs found
+        // Considering the topDocs can be empty if no result is found on the shard therefore we need iterate over all the shards .
+        for (CompoundTopDocs compoundTopDocs : queryTopDocs) {
+            if (compoundTopDocs == null) {
+                throw new IllegalArgumentException("CompoundTopDocs cannot be null in the normalization process");
+            }
+            if (checkIfTopFieldDocExists(compoundTopDocs.getTopDocs())) {
+                return compoundTopDocs.getTopDocs().toArray(new TopFieldDocs[0]);
+            }
+        }
+        return new TopFieldDocs[0];
+    }
+
+    private static boolean checkIfTopFieldDocExists(List<TopDocs> topDocs) {
+        // topDocs can be empty if no results found in the shard
+        if (topDocs == null || topDocs.isEmpty()) {
+            return false;
+        }
+        for (TopDocs topDoc : topDocs) {
+            if (topDoc != null && topDoc instanceof TopFieldDocs) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Creates Sort object from topFieldsDocs fields.
+     * It is necessary to widen the SortField.Type to maximum byte size for merging sorted docs.
+     * Different indices might have different types. This will avoid user to do re-index of data
+     * in case of mapping field change for newly indexed data.
+     * This will support Int to Long and Float to Double.
+     * Earlier widening of type was taken care in IndexNumericFieldData, but since we now want to
+     * support sort optimization, we removed type widening there and taking care here during merging.
+     * More details here https://github.com/opensearch-project/OpenSearch/issues/6326
+     */
+    private static Sort createSort(TopFieldDocs[] topFieldDocs) {
+        if (topFieldDocs == null || topFieldDocs[0] == null) {
```

Comment:
why are we only checking first value?

Reply:
Because, this check is not even needed because this condition would never happen. I have kept it here because later when I will add the additional unit tests then it doesn't fail.